crypto/types.go

```diff
@@ -110,20 +111,22 @@ type Signature []byte
 // It allows a function caller differentiate unexpected program errors from errors caused by
 // invalid inputs.
 type InvalidInputsError struct {
-	message string
+	err error
 }
 
-// newInvalidInputsError constructs a new InvalidInputsError
-func newInvalidInputsError(msg string, args ...interface{}) error {
-	return &InvalidInputsError{message: fmt.Sprintf(msg, args...)}
+// invalidInputsErrorf constructs a new InvalidInputsError
+func invalidInputsErrorf(msg string, args ...interface{}) error {
```

Comment:
Just curious why the name change here? Using `new` to indicate a constructor makes sense and matches the other similar error constructors we have, eg: https://github.com/onflow/flow-go/blob/c3d239ee1534db5430ddd8b9a747f40063ad19ed/engine/errors.go#L25

Reply:
I just went for a more compact name, and it follows other libraries like `fmt.Errof` instead of `fmt.NewErrorf`.